frontend/public/components/factory/list.jsx

```diff
@@ -99,6 +99,12 @@ const listFilters = {
 
     let status = routeStatus(route);
     return statuses.selected.has(status) || !_.includes(statuses.all, status);
+  },
+  'secret-type': (types, secret) => {
+    if (!types || !types.selected || !types.selected.size) {
+      return true;
+    }
+    return types.selected.has(secret.type);
```

Comment:
If I create a secret with a type that isn't one of the ones we know about, what happens? (It's not blocked by the API.)